src/agent/srcview/tests/srcview.rs

```diff
@@ -0,0 +1,67 @@
+use std::env;
+use std::path::PathBuf;
+
+use srcview::{ModOff, SrcLine, SrcView};
+
+fn test_srcview() -> SrcView {
+    let root = env::var("CARGO_MANIFEST_DIR").unwrap();
+    let pdb_path: PathBuf = [&root, "res", "example.pdb"].iter().collect();
```

Comment:
I think we want to avoid committing binary files (here, the `.pdb`) to the repo. This is a pain for your tests, but we don't have a great solution for this yet. Our approach has been to build binary artifacts on-demand in CI, but that doesn't work for some of these _very_ binary-oriented coverage scenarios.

There's one existing example where we have tests that use a fixed binary. We gate with a build flag: https://github.com/microsoft/onefuzz/blob/main/src/agent/onefuzz-agent/src/tasks/fuzz/supervisor.rs#L290

And then use an env var to look up the binary: https://github.com/microsoft/onefuzz/blob/main/src/agent/onefuzz-agent/src/tasks/fuzz/supervisor.rs#L298

That integration test is currently disabled. We need to revisit binary test artifact management for it and some other tests, too. For example, I have done some manual binary analysis for sancov-related things, and I'd like to have regular integration tests that require fixed binaries.

For now, WDYT about just gating the PDB-dependent integration tests behind a feature flag, and opening an issue to properly address this?

Reply:
(Opened: #1143)

Reply:
I can do this, but I'm not sure I understand the suggestion.

I can feature gate the PDB dependent tests, but then I'm unclear if the PDB still gets checked in:
- If the PDB is checked in, I'm not sure what purpose the feature gate has as the PDB still ends up in the repo.
- If the PDB is not checked in, then I very much dislike it and would prefer to delete the tests -- the tests are only meaningful when combined with the reference PDB, and without it I would say are worse than not existing.

While I'm pretty sure this is understood, without the PDB there would effectively be no tests beyond compilation in the doc examples.

Reply:
I was just thinking we should _not_ check in the PDB, but add a binary artifact integration test feed in a future PR (closing #1143), and properly re-enable your tests then. The PDB you use would then get added to whatever feed we use for binary test artifacts. So, we would have some unused (in CI) tests for a bit, but not forever. And in the interim, tests that can only be run manually still seem valuable. We'd probably just want to document e.g. the SHA256 hash of the test PDB.

Does something like that seem reasonable?

Reply:
Sure, the only part that's still unclear to me is where the PDB lives in the interim.

Reply:
In the interim, I was thinking just on your workstation? Open to ideas. And if you really hate it, we could totally just remove the tests for now, and add them back later. Just trying to reduce tedium on your end.